plugin/app/feishu/feishu.go

```diff
@@ -0,0 +1,426 @@
+	if err != nil {
+		return nil, err
+	}
+	payload := createApprovalInstanceReq{
+		ApprovalCode: approvalCode,
+		Form:         formValue,
+		NodeApproverOpenIDList: []struct {
+			Key   string   `json:"key"`
+			Value []string `json:"value"`
+		}{
+			{
+				Key:   "approve-here",
+				Value: []string{approverID},
+			},
+		},
+		OpenID: creatorID,
+	}
+	body, err := json.Marshal(payload)
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest("POST", "https://open.feishu.cn/open-apis/approval/v4/instances", bytes.NewBuffer(body))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+p.Token)
+	resp, err := p.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var response ExternalApprovalResponse
+	if err := json.Unmarshal(b, &response); err != nil {
+		return nil, err
+	}
+
+	if response.Code != 0 {
+		return nil, errors.Errorf("failed to create approval instance, %s", response.Msg)
+	}
+
+	return &response, nil
+}
+
+// GetExternalApproval gets the status of an approval instance.
+func (p *feishuProvider) GetExternalApproval(instanceCode string) (*GetExternalApprovalResponse, error) {
+	url := fmt.Sprintf("https://open.feishu.cn/open-apis/approval/v4/instances/%s", instanceCode)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Authorization", "Bearer "+p.Token)
+	resp, err := p.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	var response GetExternalApprovalResponse
+	if err := json.Unmarshal(b, &response); err != nil {
+		return nil, err
+	}
+	if response.Code != 0 {
+		return nil, errors.Errorf("failed to get approval instance, %s", response.Msg)
+	}
+
+	return &response, nil
+}
+
+// CancelExternalApproval cancels an approval instance.
+func (p *feishuProvider) CancelExternalApproval(approvalCode, instanceCode, userID string) (*CancelExternalApprovalResponse, error) {
+	body := strings.NewReader(fmt.Sprintf(cancelExternalApprovalReq, approvalCode, instanceCode, userID))
+	req, err := http.NewRequest("POST", "https://open.feishu.cn/open-apis/approval/v4/instances/cancel", body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+p.Token)
+	resp, err := p.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var response CancelExternalApprovalResponse
+	if err := json.Unmarshal(b, &response); err != nil {
+		return nil, err
+	}
+
+	if response.Code != 0 {
+		return nil, errors.Errorf("failed to create approval instance, %s", response.Msg)
+	}
+
+	return &response, nil
```

Comment:
Same here. We should return the relevant information instead of the whole response. The caller should not be responsible to transforming the feishu implementation details.